spec/classes/lib/ccreds_spec.rb

```diff
@@ -1,146 +1,198 @@
 require 'spec_helper'
-describe 'pam::lib::ccreds', :type => :class do
-  context 'on a RedHat OS' do
-    let(:facts) {
-      {
-        :osfamily                   => 'RedHat',
-        :operatingsystemmajrelease  => '5',
-      }
-    }
-    context 'with no parameters' do
-      it { should contain_package('pam_ccreds').with(
-        'ensure'  => 'installed'
-      ) }
-    end
-    context 'when absent' do
-      let :params do
-          {
-            :ensure  => 'absent'
-          }
+describe 'pam::lib::ccreds' do
+
+  platforms = {
```

Comment:
defining all the platforms we support and some keys that can be mapped to facts that need to be set.